jina/drivers/rank/__init__.py

```diff
@@ -52,8 +52,8 @@ def _apply_all(self, docs: 'DocumentSet', context_doc: 'Document', *args,
         # if at the top-level already, no need to aggregate further
         query_meta = context_doc.get_attrs(*self.exec.required_keys)
 
-        old_match_scores = {int(match.id): match.score.value for match in docs}
-        match_meta = {int(match.id): match.get_attrs(*self.exec.required_keys) for match in docs}
+        old_match_scores = {str(match.id): match.score.value for match in docs}
```

Comment:
Why do you convert to string if you then convert back to `UniqueId`?

Reply:
Hey @cristianmtr,
I now removed the string-uniqueid-conversion in (e204475c658d19)[https://github.com/janandreschweiger/jina/commit/e204475c658d19b937d3a18c3cb175ac55533d49]. Unfortunately, the jinad-test (ubuntu-latest, 3.7) now fails. I tried several things, but I cannot fix this error. Could you maybe have a look on it?

Reply:
Hey @janandreschweiger,

have u rebased on master? these tests should work if you realign, since jinad has recently been updated

Reply:
Thanks @JoanFM, I will try that after adding some unit tests.

Reply:
Hey @JoanFM the intergration test still fails. Should I revert my changes and use strings?

Reply:
Hey @janandreschweiger,

No, do not revert it. We will check what can be the reason, but one colleague is doing some refactoring on that. We will investigate what can be the problem and see how it can be fixed. For now let's focus on having unit and integration tests working